env/activate.fish

```diff
@@ -1,48 +1,39 @@
-function deactivate -d "Exit virtual environment and return to normal shell environment"
-    if test -n "$_OLD_VIRTUAL_PATH"
+function deactivate
+    if set -q _OLD_VIRTUAL_PATH
         set -gx PATH $_OLD_VIRTUAL_PATH
         set -e _OLD_VIRTUAL_PATH
     end
-    if test -n "$_OLD_VIRTUAL_PYTHONHOME"
+
+    if set -q _OLD_VIRTUAL_PYTHONHOME
         set -gx PYTHONHOME $_OLD_VIRTUAL_PYTHONHOME
         set -e _OLD_VIRTUAL_PYTHONHOME
     end
-    if test -n "$_OLD_FISH_PROMPT_OVERRIDE"
+
+    if set -q _OLD_FISH_PROMPT_OVERRIDE
+        functions -e fish_prompt
+        functions -c _old_fish_prompt fish_prompt
+        functions -e _old_fish_prompt
         set -e _OLD_FISH_PROMPT_OVERRIDE
-        if functions -q _old_fish_prompt
-            functions -e fish_prompt
-            functions -c _old_fish_prompt fish_prompt
-            functions -e _old_fish_prompt
-        end
-    end
-    if test "$argv[1]" != nondestructive
-        set -e TTFORGE_TOOLCHAIN_DIR
-        set -e TTFORGE_PYTHON_VERSION
-        set -e TTFORGE_VENV_DIR
-        set -e TTMLIR_TOOLCHAIN_DIR
-        set -e TTMLIR_VENV_DIR
-        set -e TTMLIR_ENV_ACTIVATED
-        set -e ARCH_NAME
     end
+
     set -e VIRTUAL_ENV
     set -e VIRTUAL_ENV_PROMPT
-    if test "$argv[1]" != nondestructive
-        # Self-destruct!
-        functions -e deactivate
-    end
+    set -e TTFORGE_VENV_DIR
+    set -e TTMLIR_ENV_ACTIVATED
+
+    functions -e deactivate
 end
 
 deactivate nondestructive
 
-set -gx TTFORGE_TOOLCHAIN_DIR (test -n "$TTFORGE_TOOLCHAIN_DIR"; and echo "$TTFORGE_TOOLCHAIN_DIR"; or echo "/opt/ttforge-toolchain")
-set -gx TTFORGE_PYTHON_VERSION (test -n "$TTFORGE_PYTHON_VERSION"; and echo "$TTFORGE_PYTHON_VERSION"; or echo "python3.10")
-set -gx TTFORGE_VENV_DIR (test -n "$TTFORGE_VENV_DIR"; and echo "$TTFORGE_VENV_DIR"; or echo "$TTFORGE_TOOLCHAIN_DIR/venv")
-set -gx TTMLIR_TOOLCHAIN_DIR (test -n "$TTMLIR_TOOLCHAIN_DIR"; and echo "$TTMLIR_TOOLCHAIN_DIR"; or echo "/opt/ttmlir-toolchain")
-set -gx TTMLIR_VENV_DIR (test -n "$TTMLIR_VENV_DIR"; and echo "$TTMLIR_VENV_DIR"; or echo "$TTMLIR_TOOLCHAIN_DIR/venv")
+set -gx TTFORGE_TOOLCHAIN_DIR $TTFORGE_TOOLCHAIN_DIR "/opt/ttforge-toolchain"[1]
```

Comment:
@Gogopex 

This syntax didn't do the right thing for me. it just ended up giving me a variable with the literal `/opt/ttforge-toolchain[1]` in it (or an array with the existing value and that literal). I didn't see a construct like this in the fish manual.